sqlglot/dialects/tsql.py

```diff
@@ -37,11 +106,52 @@ class Parser(Parser):
             "CHARINDEX": exp.StrPosition.from_arg_list,
         }
 
-        def _parse_convert(self):
+        def _parse_convert(self, strict):
             to = self._parse_types()
             self._match(TokenType.COMMA)
             this = self._parse_field()
-            return self.expression(exp.Cast, this=this, to=to)
+
+            var_length_datatype = [
+                DataType.Type.NVARCHAR,
+                DataType.Type.VARCHAR,
+                DataType.Type.CHAR,
+                DataType.Type.NCHAR,
+            ]
+
+            # Retrieve length of datatype and override to default if not specified
+            if list_get(to.expressions, 0) is None and to.this in var_length_datatype:
+                to = exp.DataType.build(to.this, expressions=[exp.Literal(this="30", is_string=False)], nested=False)
+
+            # Check whether a conversion with format is applicable
+            if self._match(TokenType.COMMA):
+                format_norm = TSQL.convert_format_mapping[int(self._parse_number().this)]
+
+                # Check whether the convert entails a string to date format
+                if to.this in [DataType.Type.DATE]:
+                    return self.expression(
+                        exp.StrToDate, this=this, format=exp.Literal(this=format_norm, is_string=True)
+                    )
+                # Check whether the convert entails a string to datetime format
+                elif to.this in [DataType.Type.DATETIME]:
+                    return self.expression(
+                        exp.StrToTime, this=this, format=exp.Literal(this=format_norm, is_string=True)
+                    )
+                # Check whether the convert entails a date to string format
+                elif to.this in var_length_datatype:
+                    return self.expression(
+                        exp.Cast if strict else exp.TryCast,
+                        to=to,
+                        this=self.expression(
+                            exp.TimeToStr, this=this, format=exp.Literal(this=format_norm, is_string=True)
```

Comment:
change all to Literal.string

Reply:
Changed to Literal.string and is included in the format_norm object now.